client/subscription-product-onboarding/modal.js

```diff
@@ -57,6 +57,11 @@ const SubscriptionProductOnboardingModal = () => {
 		return null;
 	}
 
+	wcpayTracks.recordEvent(
+		wcpayTracks.events
+			.SUBSCRIPTIONS_ACCOUNT_NOT_CONNECTED_PRODUCT_MODAL_VIEW
+	);
```

Comment:
Hey @haszari, we probably want to move this logic into a new [effect](https://reactjs.org/docs/hooks-effect.html).

Example here → https://github.com/Automattic/woocommerce-payments/blob/3817794/client/subscriptions-empty-state/index.js#L106-L113

Reply:
Good point, thanks for catching that! I agree, this should be wrapped in an effect so it fires only once, [i.e. a "mount" effect](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects).

Fixed in this commit: e4df890d4ca81cb665dcddc1933581b63fd4f09a

Note to do this I needed to factor out the modal content into a new component, and add the mount effect there. That was necessary because the event should only fire if `isOpen`, and conditionals and effects don't mix. Let me know if there's a better way to do this :)